ex7/k_furuta/main.py

```diff
@@ -0,0 +1,236 @@
+    MaxPooling2D,
+)
+from tensorflow.keras.models import Sequential, load_model
+from tensorflow.keras.optimizers import SGD
+from tensorflow.keras.utils import to_categorical
+
+
+def my_CNN(input_shape, output_dim):
+    """CNNモデルの構築.
+
+    Args:
+        input_shape: 入力の形
+        output_dim: 出力次元
+    Returns:
+        model: 定義済みモデル
+    """
+    model = Sequential()
+    model.add(
+        Conv2D(
+            32,
+            (3, 3),
+            activation="relu",
+            input_shape=(input_shape[0], input_shape[0], 1),
+        )
+    )
+    model.add(MaxPooling2D((2, 2)))
+    model.add(Conv2D(64, (3, 3), activation="relu"))
+    model.add(MaxPooling2D((2, 2)))
+    model.add(Conv2D(128, (3, 3), activation="relu"))
+    model.add(MaxPooling2D((2, 2)))
+    model.add(Flatten())
+    model.add(Dense(1024, activation="relu"))
+    model.add(Dropout(0.5)),
+    model.add(Dense(128, activation="relu"))
+    model.add(Dropout(0.5)),
+    model.add(Dense(output_dim, activation="softmax"))
+    model.summary()  # モデル構成の表示
+    return model
+
+
+def load_data(path):
+    """ファイルへのパスからデータをロードする関数."""
+    return librosa.load(path)[0]
+
+
+def feature_extraction(path_list):
+    """wavファイルのリストから特徴抽出を行い，リストで返す.
+
+    扱う特徴量はメルスペクトログラム
+    Args:
+        path_list: 特徴抽出するファイルのパスリスト
+    Returns:
+        features: 特徴量
+    """
+    data = list(map(load_data, path_list))
+    features = []
+    for y in data:
+        mel_spectrogram = librosa.feature.melspectrogram(y=y)  # メルスペクトログラム
+        mel_spectrogram_dB = librosa.power_to_db(
+            mel_spectrogram, ref=np.max
+        )  # デシベル単位に変換
+        mel_spectrogram_dB_resize = Image.fromarray(mel_spectrogram_dB).resize(
+            (128, 128)
+        )  # 時間方向に拡大して正方形画像とする
+        mel_spectrogram_dB_square = np.asarray(
+            mel_spectrogram_dB_resize
+        )  # np.ndarrayに戻す
+        features.append(mel_spectrogram_dB_square)  # featuresにappend
+    features = np.stack(features)  # shape=(sample, 128, 128)になる
+    return features
+
+
+def plot_confusion_matrix(predict, ground_truth, title=None, cmap=plt.cm.Blues):
+    """予測結果の混合行列をプロット.
+
+    Args:
+        predict: 予測結果
+        ground_truth: 正解ラベル
+        title: グラフタイトル
+        cmap: 混合行列の色
+    Returns:
+        Nothing
+    """
+    cm = confusion_matrix(ground_truth, predict)
+    plt.figure()
+    plt.imshow(cm, interpolation="nearest", cmap=cmap)
+    plt.title(title)
+    plt.colorbar()
+    plt.tight_layout()
+    plt.ylabel("Ground truth")
+    plt.xlabel("Predicted")
+    plt.savefig("result.png")
+
+
+def write_result(paths, outputs):
+    """結果をcsvファイルで保存する.
+
+    Args:
+        paths: テストする音声ファイルリスト
+        outputs:
+    Returns:
+        Nothing
+    """
+    with open("result.csv", "w") as f:
+        f.write("path,output\n")
+        assert len(paths) == len(outputs)
+        for path, output in zip(paths, outputs):
+            f.write("{path},{output}\n".format(path=path, output=output))
```

Comment:
```suggestion
            f.write(f"{path},{output}\n")
```
f-stringという機能を使うと，こんな感じで文字列中に直接変数を埋め込むことができてシンプルに記述できるのでオススメです